contracts/Strategy.sol

```diff
@@ -38,49 +26,100 @@ contract Strategy is BaseStrategy {
-        ISwapRouter(0xE592427A0AEce92De3Edee1F18E0157C05861564);
+    IUniswapV2Router private constant SUSHI_V2_ROUTER =
+        IUniswapV2Router(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F);
+    IUniswapV3Router private constant UNI_V3_ROUTER =
+        IUniswapV3Router(0xE592427A0AEce92De3Edee1F18E0157C05861564);
 
     // OPS State Variables
-    uint24 public stkAaveToAaveSwapFee = 10000;
-
-    uint16 private constant referral = 7; // Yearn's aave referral code
+    uint24 public aaveToStkAaveSwapFee;
+    uint256 public stkAaveDiscountBps;
+    bool public forceCooldown;
+    uint256 public dustThreshold;
 
     uint256 private constant MAX_BPS = 1e4;
 
     constructor(address _vault) public BaseStrategyInitializable(_vault) {
         _initializeThis();
     }
 
+    function initialize(
+        address _vault,
+        address _strategist,
+        address _rewards,
+        address _keeper
+    ) external override {
+        _initialize(_vault, _strategist, _rewards, _keeper);
+        _initializeThis();
+    }
+
     function _initializeThis() internal {
         require(address(want) == address(aave));
 
-        stkAaveToAaveSwapFee = 10000;
+        aaveToStkAaveSwapFee = 3000;
+        stkAaveDiscountBps = 150;
+        forceCooldown = false;
+        dustThreshold = 1e13;
 
         // approve swap router spend
         approveMaxSpend(address(stkAave), address(UNI_V3_ROUTER));
         approveMaxSpend(aave, address(UNI_V3_ROUTER));
     }
 
-    function setSwapFee(
-        uint24 _stkAaveToAaveSwapFee
-    ) external onlyVaultManagers {
-        stkAaveToAaveSwapFee = _stkAaveToAaveSwapFee;
+    function setSwapFee(uint24 _aaveToStkAaveSwapFee)
+        external
+        onlyVaultManagers
+    {
+        require(
+            _aaveToStkAaveSwapFee == 500 ||
+                _aaveToStkAaveSwapFee == 3000 ||
+                _aaveToStkAaveSwapFee == 10000
+        );
+        aaveToStkAaveSwapFee = _aaveToStkAaveSwapFee;
+    }
+
+    function setDiscount(uint256 _stkAaveDiscountBps)
+        external
+        onlyVaultManagers
+    {
+        require(_stkAaveDiscountBps <= MAX_BPS); // dev: >MAX_BPS
+        stkAaveDiscountBps = _stkAaveDiscountBps;
+    }
+
+    function setForceCooldown(bool _forceCooldown) external onlyVaultManagers {
+        forceCooldown = _forceCooldown;
+    }
+
+    function setDustThreshold(uint256 _dustThreshold)
+        external
+        onlyVaultManagers
+    {
+        dustThreshold = _dustThreshold;
     }
 
     function name() external view override returns (string memory) {
         return "StrategyStkAaveCooldown";
     }
 
     function estimatedTotalAssets() public view override returns (uint256) {
-        return balanceOfWant().add(balanceOfStkAave());
+        uint256 discountedStkAave = MAX_BPS
+            .sub(stkAaveDiscountBps)
+            .mul(balanceOfStkAave())
+            .div(MAX_BPS);
+        return balanceOfWant().add(discountedStkAave);
```

Comment:
just hardcode `MAX_BPS` here instead of having an additional variable?
Can you add a comment on why the `sub(stkAaveDiscountBps)` for future generations? :D